packages/liveblocks-yjs/src/doc.ts

```diff
@@ -40,19 +44,26 @@ export default class yDocHandler extends Observable<unknown> {
     update,
     stateVector,
     readOnly,
+    v2,
   }: {
     update: Uint8Array;
     stateVector: string | null;
     readOnly: boolean;
+    v2?: boolean;
   }): void => {
-    // apply update from the server
-    Y.applyUpdate(this.doc, update, "backend");
+    // apply update from the server, updates from the server can be v1 or v2
+    const applyUpdate = v2 ? Y.applyUpdateV2 : Y.applyUpdate;
+    applyUpdate(this.doc, update, "backend");
     // if this update is the result of a fetch, the state vector is included
     if (stateVector) {
       if (!readOnly) {
         // Use server state to calculate a diff and send it
         try {
-          const localUpdate = Y.encodeStateAsUpdate(
+          // send v1 or v2update according to client option
+          const encodeUpdate = this.useV2Encoding
+            ? Y.encodeStateAsUpdateV2
+            : Y.encodeStateAsUpdate;
+          const localUpdate = encodeUpdate(
```

Comment:
Mostly for my own knowledge but wasn't it more safe to handle the server update on the same encoding option for every operation? I mean are we sure it's safe to receive a v2 update and potentially make an encode operation on v1 as `useV2Encoding` can be `false`? 🤔 
I read in your back-end PR that messages can be v1 or v2 and won't affect any functionality but I'm wondering if wouldn't be more robust and safe to strictly make the encode operation if `v2 === true && this.useV2Encoding === true` to avoid any dissonances no? 🤔

Reply:
Sending updates is completely independent of receiving them. This specific line of code happens to be in a function where we perform a "sync" if we receive state vector which tells our client to respond with an update containing anything since that state vector. State vectors are unaffected by v1 or v2 encoding.